funcframework/framework_test.go

```diff
@@ -121,6 +121,132 @@ type eventData struct {
 	Data string `json:"data"`
 }
 
+func TestRegisterTypedFunction(t *testing.T) {
+	var tests = []struct {
+		name       string
+		path       string
+		body       []byte
+		fn         func(customStruct) customStruct
+		target     string
+		status     int
+		header     string
+		ceHeaders  map[string]string
+		wantResp   string
+		wantStderr string
+	}{
```

Comment:
We probably also want a test case for
- built in types
- type without json tags
- returning an error
- any violations of number of inputs/returns